### PR TITLE
Add S3 bucket tags and encrypt S3 bucket

### PR DESCRIPTION
**Pull request by njuettner**
Issues: https://github.com/giantswarm/roadmap/issues/984 https://github.com/giantswarm/roadmap/issues/985

- Enable encryption for OIDC S3 bucket.
- Add GiantSwarm tags to OIDC S3 bucket.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/aws/scope/cluster.go [hunk 1]
```diff
@@ -135,6 +141,11 @@ func (s *ClusterScope) ClusterNamespace() string {
 	return s.clusterNamespace
 }
 
+// Region returns the name of the installation where the cluster object is located.
```
Comment (njuettner) [inline pkg/aws/scope/cluster.go:144]:
```suggestion
// Installation returns the name of the installation where the cluster object is located.
```

File: main.go [hunk 2]
```diff
@@ -47,6 +47,8 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
+	var installation = os.Getenv("INSTALLATION")
```
Comment (whites11) [inline main.go:50]: why an env and not a flag like other fields?

**Review by whites11 — APPROVED** [review]
LGTM

File: main.go (hunk 2 above)
Comment (njuettner) [reply main.go]: I'm an idiot, I haven't pushed the change 🤦🏻

File: helm/irsa-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -34,7 +34,7 @@ spec:
         - name: INSTALLATION
```
Comment (whites11) [inline helm/irsa-operator/templates/deployment.yaml:34]: I guess you can remove this now

**Pull request merged**